src/Components/Server/src/LoggerExtensions.cs

```diff
@@ -9,13 +9,31 @@ namespace Microsoft.AspNetCore.Components.Server
     internal static class LoggerExtensions
     {
         private static readonly Action<ILogger, string, Exception> _unhandledExceptionRenderingComponent;
+        private static readonly Action<ILogger, string, Exception> _disposingCircuit;
+        private static readonly Action<ILogger, string, Exception> _unhandledExceptionDisposingCircuitHost;
+        private static readonly Action<ILogger, string, Exception> _unhandledExceptionInvokingCircuitHandler;
```

Comment:
I didn't realise that we'd already started down this path, but I'm going to complain about it. Using this kind of organization in MVC has led to serveral bugs, generally due to improper re-use of logging extension methods - because they're all in one big pile.

Here's a pattern that I have used in a few projects that makes the logging scoped to the types that use it, and it's about as terse.  https://github.com/aspnet/Extensions/blob/master/src/HttpClientFactory/Http/src/DefaultHttpClientFactory.cs#L333

Reply:
Everything is solved with more nested classes.